ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite_core.cc

```diff
@@ -411,27 +501,58 @@ TFLiteCore::setInputTensorDim (const GstTensorsInfo * info)
       /* the order of dimension is reversed at CAPS negotiation */
       for (int idx = 0; idx < rank; ++idx) {
         /** check overflow when storing uint32_t in int container */
-        if (tensor_info->dimension[rank - idx - 1] > INT_MAX)
-          return -ERANGE;
+        if (tensor_info->dimension[rank - idx - 1] > INT_MAX) {
+          err = -ERANGE;
+          goto out;
+        }
         dims[idx] = tensor_info->dimension[rank - idx - 1];
       }
 
-      status = interpreter->ResizeInputTensor(input_idx_list[tensor_idx], dims);
+      status = interpreter.get()->ResizeInputTensor(input_idx_list[tensor_idx], dims);
       if (status != kTfLiteOk)
         continue;
 
       break;
     }
 
     /** return error when none of the ranks worked */
-    if (status != kTfLiteOk)
-      return -EINVAL;
-
+    if (status != kTfLiteOk) {
+      err = -EINVAL;
+      goto out;
+    }
   }
 
-  status = interpreter->AllocateTensors();
+  status = interpreter.get()->AllocateTensors();
   if (status != kTfLiteOk)
-    return -EINVAL;
+    err = -EINVAL;
+
+out:
+  interpreter.unlock ();
+
+  return err;
+}
+
+/**
+ * @brief	reload a model
+ * @param	tflite	: the class object
+ * @param[in] model_path : the path of model file
+ * @return 0 if OK. non-zero if error.
+ * @note reloadModel() is asynchronously called with other callbacks
+ */
+int
+TFLiteCore::reloadModel (const char * model_path)
+{
+  interpreter_sub.lock();
+
+  /* load a model into the sub interp; it's indepedendent with main one's activities */
+  interpreter_sub.loadModel (model_path, use_nnapi);
+
+  /* assign sub interp to main one; it waits if main interp is already doing invoke() */
+  interpreter.lock();
+  interpreter = interpreter_sub;
+  interpreter.unlock();
+
+  interpreter_sub.unlock();
```

Comment:
You are double-unlocking interpreter_sub and keeping it locked used-to-be-interpreter. (because of ```interpreter = interpreter_sub;```)

Reply:
This assignment does not affect the lock because the assignment operator was overloaded.
It just replaced the internal interpreter instance.

Reply:
Then, don't override an operator = with "model/interpreter assignment", it's not readable.
Recalling OO experts: how do you think about it?: @jaeyun-jung @again4you

Reply:
Got it. I'll make an explicit function to perform this action.